File: tests-playwright/src/pages/api-explorer/api-explorer-page.ts
import { expect, type Locator, type Page } from '@playwright/test'
import { routes } from '../../constants/routes'
import { BasePage } from '../base-page'

/** Page object for the API explorer: request groups and the response panel. */
export class ApiExplorerPage extends BasePage {
  readonly root: Locator
  readonly coinsGroup: Locator
  readonly simulationsGroup: Locator
  readonly emptyResponse: Locator

  constructor(page: Page) {
    super(page)
    this.root = page.getByTestId('api-explorer-page')
    this.coinsGroup = page.getByTestId('api-group-coins')
    this.simulationsGroup = page.getByTestId('api-group-simulations')
    this.emptyResponse = page.getByTestId('api-response-empty')
  }

  async open(): Promise<void> {
    await this.goto(routes.apiExplorer)
    await expect(this.root).toBeVisible()
  }

  async verifyShellReady(): Promise<void> {
    await expect(this.coinsGroup).toBeVisible()
    await expect(this.simulationsGroup).toBeVisible()
    await expect(this.emptyResponse).toBeVisible()
  }
}

File: tests-selenium/src/test/java/com/coinpulse/selenium/pages/ApiExplorerPage.java
package com.coinpulse.selenium.pages;

import com.coinpulse.selenium.constants.Routes;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

/**
 * API Explorer page object; selectors mirror the Cypress page object.
 */
public class ApiExplorerPage extends BasePage {

    private static final By PAGE = byTestId("api-explorer-page");
    private static final By GROUP_COINS = byTestId("api-group-coins");
    private static final By GROUP_SIMULATIONS = byTestId("api-group-simulations");
    private static final By RESPONSE_EMPTY = byTestId("api-response-empty");

    public ApiExplorerPage(WebDriver driver) {
        super(driver);
    }

    public void open() {
        openPath(Routes.API_EXPLORER);
        awaitVisible(PAGE);
    }

    public void verifyShellReady() {
        awaitVisible(GROUP_COINS);
        awaitVisible(GROUP_SIMULATIONS);
        awaitVisible(RESPONSE_EMPTY);
    }
}
